bindata/network/ovn-kubernetes/common/ipsec-host.yaml

```diff
@@ -397,6 +333,125 @@ spec:
+          # that will be used to talk to the API
+
+
+          # Wait for cert file
+          retries=0
+          tries=20
+          key_cert="/etc/ovn/ovnkube-node-certs/ovnkube-client-current.pem"
+          while [ ! -f "${key_cert}" ]; do
+            (( retries += 1 ))
+            if [[ "${retries}" -gt ${tries} ]]; then
+              echo "$(date -Iseconds) - ERROR - ${key_cert} not found"
+              return 1
+            fi
+            sleep 1
+          done
+
+          cat << EOF > /var/run/ovnkube-kubeconfig
+          apiVersion: v1
+          clusters:
+            - cluster:
+                certificate-authority: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+                server: {{.K8S_APISERVER}}
+              name: default-cluster
+          contexts:
+            - context:
+                cluster: default-cluster
+                namespace: default
+                user: default-auth
+              name: default-context
+          current-context: default-context
+          kind: Config
+          preferences: {}
+          users:
+            - name: default-auth
+              user:
+                client-certificate: /etc/ovn/ovnkube-node-certs/ovnkube-client-current.pem
+                client-key: /etc/ovn/ovnkube-node-certs/ovnkube-client-current.pem
+          EOF
+          export KUBECONFIG=/var/run/ovnkube-kubeconfig
+{{ end }}
+
+          # It is safe to flush xfrm states and policies and delete openshift.conf
+          # file when east-west ipsec is disabled. This fixes a race condition when
+          # ovs-monitor-ipsec is not fast enough to notice ipsec config change and
+          # delete entries before it's being killed.
+          # Since it's cleaning up all xfrm states and policies, it may cause slight
+          # interruption until ipsec is restarted in case of external ipsec config.
+          # We must do this before killing ovs-monitor-ipsec script, otherwise
+          # preStop hook doesn't get a chance to run it because ovn-ipsec container
+          # is abruptly terminated.
+          # When east-west ipsec is not disabled, then do not flush xfrm states and
+          # policies in order to maintain traffic flows during container restart.
+          ipsecflush() {
+            if [ "$(kubectl get networks.operator.openshift.io cluster -ojsonpath='{.spec.defaultNetwork.ovnKubernetesConfig.ipsecConfig.mode}')" != "Full" ] || \
+               [ "$(kubectl get networks.operator.openshift.io cluster -ojsonpath='{.spec.defaultNetwork.ovnKubernetesConfig.ipsecConfig}')" != "{}" ]; then
+              ip x s flush
+              ip x p flush
+              rm -f /etc/ipsec.d/openshift.conf
+              # since pluto is on the host, we need to restart it after the flush
+              chroot /proc/1/root ipsec restart
+            fi
+          }
+
+          # Function to handle SIGTERM
+          cleanup() {
```

Comment:
when a pod is deleted and all containers receive sigterm, i guess this will go ahead and flush before ovs-monitor-ipsec exits. Maybe you should wait a maximum of like 15 seconds or something for ovs-monitor script to exit before you do the ipsec cleanup. I think default termination grace period is 30 seconds, so waiting 15 should be ok.

Reply:
yes @trozet , good find. done.